x-pack/plugins/cases/server/client/cases/bulk_get.ts

```diff
@@ -0,0 +1,159 @@
+    );
+
+    throwErrorIfCaseIdsReachTheLimit(request.ids);
+    throwErrorIfFieldsAreInvalid(fields);
+
+    const cases = await caseService.getCases({ caseIds: request.ids, fields });
+
+    const [validCases, soBulkGetErrors] = partition(
+      cases.saved_objects,
+      (caseInfo) => caseInfo.error === undefined
+    ) as [CaseSavedObject[], SOWithErrors];
+
+    const { authorized: authorizedCases, unauthorized: unauthorizedCases } =
+      await authorization.getAndEnsureAuthorizedEntities({ savedObjects: validCases });
+
+    const requestForTotals = ['totalComment', 'totalAlerts'].some(
+      (totalKey) => !fields || fields.includes(totalKey)
+    );
+
+    const commentTotals = requestForTotals
+      ? await attachmentService.getCaseCommentStats({
+          unsecuredSavedObjectsClient,
+          caseIds: authorizedCases.map((theCase) => theCase.id),
+        })
+      : new Map();
+
+    const flattenedCases = authorizedCases.map((theCase) => {
+      const { alerts, userComments } = commentTotals.get(theCase.id) ?? {
+        alerts: 0,
+        userComments: 0,
+      };
+
+      const flattenedCase = flattenCaseSavedObject({
+        savedObject: theCase,
+        totalComment: userComments,
+        totalAlerts: alerts,
+      });
+
+      if (!fields?.length) {
+        return flattenedCase;
+      }
+
+      return pick(flattenedCase, [...fields, 'id', 'version']);
+    });
+
+    const typeToEncode = getTypeForCertainFieldsFromArray(CasesResponseRt, fields);
+    const casesToReturn = typeToEncode.encode(flattenedCases);
+
+    const errors = constructErrors(soBulkGetErrors, unauthorizedCases);
+
+    return { cases: casesToReturn, errors };
+  } catch (error) {
+    const ids = params.ids ?? [];
+    throw createCaseError({
+      message: `Failed to bulk get cases: ${ids.join(', ')}: ${error}`,
+      error,
+      logger,
+    });
+  }
+};
+
+const throwErrorIfFieldsAreInvalid = (fields?: string[]) => {
+  if (!fields || fields.length === 0) {
+    return;
+  }
+
+  const typeProps = getTypeProps(CaseResponseRt) ?? {};
+  const validFields = Object.keys(typeProps);
+
+  for (const field of fields) {
+    if (!validFields.includes(field)) {
+      throw Boom.badRequest(`Field: ${field} is not supported`);
+    }
+  }
+};
+
+const throwErrorIfCaseIdsReachTheLimit = (ids: string[]) => {
+  if (ids.length > MAX_BULK_GET_CASES) {
+    throw Boom.badRequest(`Maximum request limit of ${MAX_BULK_GET_CASES} cases reached`);
+  }
+};
+
+const constructErrors = (
+  soBulkGetErrors: SOWithErrors,
+  unauthorizedCases: CaseSavedObject[]
+): CasesBulkGetResponse['errors'] => {
+  const errors: CasesBulkGetResponse['errors'] = [];
+
+  for (const soError of soBulkGetErrors) {
+    errors.push({
+      message: soError.error.message,
+      status: soError.error.statusCode,
+      caseId: soError.id,
+    });
+  }
+
+  for (const theCase of unauthorizedCases) {
+    errors.push({
+      message: 'Unauthorized',
```

Comment:
The object we return when a user is unauthorized for a retrieving an individual case looks like this:

```
{
    "statusCode": 403,
    "error": "Forbidden",
    "message": "Unauthorized to access case with owners: \"observability\""
}
```

Should we add the nested `error: 'Forbidden'` field? I don't know if it adds much 🤷‍♂️ . It would probably be helpful make the message equivalent though. So have it be `Unauthorized to access case with owner: <owner>` (I don't know why we have it as plural in the single get case scenario 🤔